File: src/main/java/io/github/chaosdave34/kitpvp/kits/ElytraKitHandler.java
package io.github.chaosdave34.kitpvp.kits;

import io.github.chaosdave34.ghutils.Utils;
import io.github.chaosdave34.kitpvp.kits.impl.elytra.*;
import lombok.Getter;

import java.util.HashMap;
import java.util.Map;

@Getter
public class ElytraKitHandler {
    private final Map<String, ElytraKit> kits = new HashMap<>();

    public static ElytraKit KNIGHT;
    public static ElytraKit SNIPER;
    public static ElytraKit PYRO;
    public static ElytraKit TANK;
    public static ElytraKit KNOCKER;
    public static ElytraKit ROCKET_LAUNCHER;
    public static ElytraKit POSEIDON;
    public static ElytraKit TELEPORTER;
    public static ElytraKit HEALER;
    public static ElytraKit CHEMIST;

    public ElytraKitHandler() {
        KNIGHT = registerKit(new KnightKit());
        SNIPER = registerKit(new SniperKit());
        PYRO = registerKit(new PyroKit());
        TANK = registerKit(new TankKit());
        KNOCKER = registerKit(new KnockerKit());
        ROCKET_LAUNCHER = registerKit(new RocketLauncherKit());
        POSEIDON = registerKit(new PoseidonKit());
        TELEPORTER = registerKit(new TeleporterKit());
        HEALER = registerKit(new HealerKit());
        CHEMIST = registerKit(new ChemistKit());
    }

    private ElytraKit registerKit(ElytraKit kit) {
        kits.put(kit.getId(), kit);
        Utils.registerEvents(kit);
        return kit;
    }
}

File: src/main/kotlin/io/github/chaosdave34/kitpvp/kits/ElytraKitHandler.kt
package io.github.chaosdave34.kitpvp.kits

import io.github.chaosdave34.ghutils.Utils
import io.github.chaosdave34.kitpvp.kits.impl.elytra.*

class ElytraKitHandler {
    val kits: MutableMap<String, ElytraKit> = mutableMapOf()

    companion object {
        lateinit var KNIGHT: ElytraKit
        lateinit var SNIPER: ElytraKit
        lateinit var PYRO: ElytraKit
        lateinit var TANK: ElytraKit
        lateinit var KNOCKER: ElytraKit
        lateinit var ROCKET_LAUNCHER: ElytraKit
        lateinit var POSEIDON: ElytraKit
        lateinit var TELEPORTER: ElytraKit
        lateinit var HEALER: ElytraKit
        lateinit var CHEMIST: ElytraKit
    }

    init {
        KNIGHT = registerKit(KnightKit())
        SNIPER = registerKit(SniperKit())
        PYRO = registerKit(PyroKit())
        TANK = registerKit(TankKit())
        KNOCKER = registerKit(KnockerKit())
        ROCKET_LAUNCHER = registerKit(RocketLauncherKit())
        POSEIDON = registerKit(PoseidonKit())
        TELEPORTER = registerKit(TeleporterKit())
        HEALER = registerKit(HealerKit())
        CHEMIST = registerKit(ChemistKit())
    }

    private fun registerKit(kit: ElytraKit): ElytraKit {
        kits[kit.id] = kit
        Utils.registerEvents(kit)
        return kit
    }
}